server/channels/store/sqlstore/channel_store.go

```diff
@@ -1190,6 +1191,15 @@ func (s SqlChannelStore) getAllChannelsQuery(opts store.ChannelSearchOpts, forCo
 		query = query.Where("c.Id NOT IN (SELECT ChannelId FROM GroupChannels WHERE GroupChannels.GroupId = ? AND GroupChannels.DeleteAt = 0)", opts.NotAssociatedToGroup)
 	}
 
+	if opts.GroupConstrained {
```

Comment:
Can we add a test case to verify the logic?

Reply:
Added via 312e7802f2b8aa7bd322d0740e5695db96c031b2